sigsep_open-unmix-pytorch_umx.md

```diff
@@ -0,0 +1,74 @@
+---
+layout: hub_detail
+background-class: hub-background
+body-class: hub
+category: researchers
+title: Open-Unmix
+summary: Reference implementation for music source separation
+image: sigsep_logo_inria.png
+author: Inria
+tags: [audio, scriptable]
+github-link: https://github.com/sigsep/open-unmix-pytorch
+github-id: sigsep/open-unmix-pytorch
+featured_image_1: sigsep_umx-diagram.png
+featured_image_2: no-image
+accelerator: cuda-optional
+---
+
+
+```bash
+# assuming you have a PyTorch >=1.6.0 installed
+pip install -q torchaudio
+```
+
+```python
+import torch
+
+# loading umxhq four target separator
+separator = torch.hub.load('sigsep/open-unmix-pytorch', 'umxhq')
+
+# audio shape (nb_samples, nb_channels, nb_timesteps)
+# first resample to the sample rate of the models, separator.sample_rate
+resampler = torchaudio.transforms.Resample(original_sample_rate, separator.sample_rate)
+audio = resampler(audio)
+audio = torch.rand((1, 2, 100000))
+estimates = separator(audio)
+# estimates.shape = (1, 4, 2, 100000)
```

Comment:
just realized the order of the instructions would read better like this:
```suggestion
# generate random audio with shape (nb_samples, nb_channels, nb_timesteps)
audio = torch.rand((1, 2, 100_000))
original_sample_rate = 48_000

# resample to the sample rate of the models, separator.sample_rate
resampler = torchaudio.transforms.Resample(original_sample_rate, separator.sample_rate)
audio = resampler(audio)
# audio.shape = (1, 2, resampled_dim)

estimates = separator(audio)
# estimates.shape = (1, 4, 2, resampled_dim)
```
thoughts?

Reply:
yes, i like that. what about making the resampling optional by setting `original_sample_rate = 44_100 = separator.model_rate` and then only resample if necessary?

```python
if original_sample_rate != separator.model_rate:
    ...
```

Reply:
That's a good point, how about simply commenting out then?
```suggestion
# generate random audio
# ... with shape (nb_samples, nb_channels, nb_timesteps)
# ... and with the same sample rate as that of the separator
audio = torch.rand((1, 2, 100_000))
original_sample_rate = separator.sample_rate

# make sure to resample the audio to models' sample rate, separator.sample_rate, if the two are different
# resampler = torchaudio.transforms.Resample(original_sample_rate, separator.sample_rate)
# audio = resampler(audio)

estimates = separator(audio)
# estimates.shape = (1, 4, 2, 100_000)
```